lib/proxy.js

```diff
@@ -57,4 +64,23 @@ function proxyHandler(req, res, configs) {
     return proxyHit;
 }
 
+
+function pathRewrite(pathRewriteConfig, configId, req) {
```

Comment:
can you see a way that we could just let users provide config data per https://www.npmjs.com/package/http-proxy#options, pass that to http-proxy and not attempt to implement features in this library at all? 

There seem to be quite a number of rewrite options in http-proxy so I'm wondering why we need to implement them here?